### PR TITLE
Graph json spec

### PR DESCRIPTION
**Pull request by ibilal**
Added all tests for JSON blobs according to relationship graph JSON specification. Corrected all lint issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "lint": "semistandard | snazzy"
   },
   "dependencies": {
+    "ajv": "^4.1.7",
```
Comment (RickyV33) [inline package.json]: This should be a dev dependency. 
@chances  can you confirm?

Comment (chances) [inline package.json]: Maybe. ajv is crazy fast and we could use it in production to validate JSON input we receive from clients

Comment (RickyV33) [inline package.json]: Oh, that's right. We discussed this already. Use ajv validation for our schema instead of manually checking

**Comment by chances** [comment]
Maybe move `lib/relationshipGraphSpec.json` to `lib/schema/relationshipGraph.json`? Not sure if we agreed on a common place for our JSON schema.

**Comment by ibilal** [comment]
I'll add it to a file under schema. I know the projectsList schema people did that. That works.

**Comment by RickyV33** [comment]
Very long lines in relationshipGraphSpec.js tests. You should format them so they don't hit that faint line that runs down the right side of webstorm/intellij.

File: test/lib/schema/relationshipGraphSpec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          body: {items: [{id: 1, name: 'story', type: 35}, {id: 2, name: 'story', type: 35}]}},
+      // test case for duplicate items
+      {title: 'should return false when JSON contains duplicate items',
+          body: {items: [{id: 1, name: 'story', type: 35}, {id: 1, name: 'story', type: 35}],
+              relationships: [{id: 1, fromItem: 1, toItem: 2, type: 36}]}}
+  ];
+  // Passing test cases for relationship graph JSON
+  let passingTestCases = [
+      {title: 'should return true when JSON contains at least 1 item and at least 1 relationship', body: {items: [{id: 34, name: 'story', type: 35}], relationships: [{id: 1, fromItem: 34, toItem: 35, type: 35}]}},
+      {title: 'should return true when JSON contains many items and many relationships with valid properties', body: {items: [{id: 34, name: 'story', type: 35},
```
Comment (RickyV33) [inline test/lib/schema/relationshipGraphSpec.js]: 57-58 should be shortened too

File: views/projects.ejs [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
 </head>
 <body>
-<h1><%= title %></h1>
-<form id="projects" method="post">
-    <input type="submit" value="Submit" />
-</form>
-<!--TODO: add error messages -->
-<!-- TODO: On a post to this page, don't change the title of the page to the error message -->
-<script src="js/project.js"></script>
+    <h2>Please click on a project below to display its Traceability Graph: </h2>
```
Comment (RickyV33) [inline views/projects.ejs]: Traceability Graph shouldn't be capitalized

**Comment by RickyV33** [comment]
LGTM besides the traceability graph capitalization

**Pull request merged**